src/main/kotlin/org/rust/ide/inspections/lints/RsNonShorthandFieldPatternsInspection.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.ide.inspections.RsProblemsHolder
+import org.rust.lang.core.psi.RsPatFieldFull
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsVisitor
+
+class RsNonShorthandFieldPatternsInspection : RsLintInspection() {
+    override fun getLint(element: PsiElement): RsLint = RsLint.NonShorthandFieldPatterns
+
+    override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean): RsVisitor = object : RsVisitor() {
+        override fun visitPatFieldFull(o: RsPatFieldFull) {
+            val identifier = o.identifier?.text
+            val binding = o.pat.text
+            if (identifier != binding) return
+
+            holder.registerProblem(
```

Comment:
Let's use `registerLintProblem` instead. It's specially created for lint inspections to take into account [lint level](https://doc.rust-lang.org/rustc/lints/levels.html) and adjust the highlighting type.
Override `org.rust.ide.inspections.lints.RsLint#toHighlightingType` to specify the highlighting type for each lint level (note, the default type for `warn` lint level is `ProblemHighlightType.WARNING`)

Reply:
Also, it would be great to add the corresponding tests, i.e. tests with `allow(non_shorthand_field_patterns)` and `deny(non_shorthand_field_patterns)` attributes